lib/phoenix_html/form.ex

```diff
@@ -185,9 +185,11 @@ defmodule Phoenix.HTML.Form do
   defstruct source: nil, impl: nil, id: nil, name: nil, data: nil,
             hidden: [], params: %{}, errors: [], options: [], index: nil
 
-  @type t :: %Form{source: Phoenix.HTML.FormData.t, name: String.t, data: %{atom => term},
-                   params: %{binary => term}, hidden: Keyword.t, options: Keyword.t,
-                   errors: Keyword.t, impl: module, id: String.t, index: nil | non_neg_integer}
+  @type t :: %Form{source: Phoenix.HTML.FormData.t, name: String.t, 
+                   data: %{atom => term} | %{String.t => term}, 
```

Comment:
`%{(atom | String.t) => term}` should be preferred as we are effectively allowing mixed keys?

Reply:
Yeah, I always feel mixed keys should be avoided, but nothing in the code speaks against it as well.